examples/structured_data/classification_with_tfdf.py

```diff
@@ -0,0 +1,726 @@
+
+First we load the data from the UCI Machine Learning Repository into a Pandas DataFrame.
+"""
+
+# Column names.
+CSV_HEADER = [
+    "age",
+    "class_of_worker",
+    "detailed_industry_recode",
+    "detailed_occupation_recode",
+    "education",
+    "wage_per_hour",
+    "enroll_in_edu_inst_last_wk",
+    "marital_stat",
+    "major_industry_code",
+    "major_occupation_code",
+    "race",
+    "hispanic_origin",
+    "sex",
+    "member_of_a_labor_union",
+    "reason_for_unemployment",
+    "full_or_part_time_employment_stat",
+    "capital_gains",
+    "capital_losses",
+    "dividends_from_stocks",
+    "tax_filer_stat",
+    "region_of_previous_residence",
+    "state_of_previous_residence",
+    "detailed_household_and_family_stat",
+    "detailed_household_summary_in_household",
+    "instance_weight",
+    "migration_code-change_in_msa",
+    "migration_code-change_in_reg",
+    "migration_code-move_within_reg",
+    "live_in_this_house_1_year_ago",
+    "migration_prev_res_in_sunbelt",
+    "num_persons_worked_for_employer",
+    "family_members_under_18",
+    "country_of_birth_father",
+    "country_of_birth_mother",
+    "country_of_birth_self",
+    "citizenship",
+    "own_business_or_self_employed",
+    "fill_inc_questionnaire_for_veteran's_admin",
+    "veterans_benefits",
+    "weeks_worked_in_year",
+    "year",
+    "income_level",
+]
+
+train_data = pd.read_csv(
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/census-income-mld/census-income.data.gz",
+    header=None,
+    names=CSV_HEADER,
+)
+
+test_data = pd.read_csv(
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/census-income-mld/census-income.test.gz",
+    header=None,
+    names=CSV_HEADER,
+)
+
+"""
+We convert the target column from string to integer.
+"""
+
+train_data["income_level"] = train_data["income_level"].apply(
+    lambda x: 0 if x == " - 50000." else 1
+)
+test_data["income_level"] = test_data["income_level"].apply(
+    lambda x: 0 if x == " - 50000." else 1
+)
+
+"""
+Now let's show the shapes of the training and test dataframes, and display some instances.
+"""
+
+print(f"Train data shape: {train_data.shape}")
+print(f"Test data shape: {test_data.shape}")
+train_data.head().T
+
+"""
+## Define dataset metadata
+
+Here, we define the metadata of the dataset that will be useful for encoding
+the input features with respect to their types.
+"""
+
+# Target feature name.
+TARGET_FEATURE_NAME = "income_level"
+# Weight column name.
+WEIGHT_COLUMN_NAME = "instance_weight"
+# Numeric feature names.
+NUMERIC_FEATURE_NAMES = [
+    "age",
+    "wage_per_hour",
+    "capital_gains",
+    "capital_losses",
+    "dividends_from_stocks",
+    "num_persons_worked_for_employer",
+    "weeks_worked_in_year",
+]
+# Categorical features and their vocabulary lists.
+CATEGORICAL_FEATURES_WITH_VOCABULARY = {
+    feature_name: sorted(
+        [str(value) for value in list(train_data[feature_name].unique())]
```

Comment:
I would cast all the bytes feature into str when loading the dataset to avoid having to handle both byte and str objects here and after.

What about creating a "process_dataframe" function that does:
- The conversion of byte to str of string values.
- The conversion from str to integer of the label values.
- ... any other dataset preprocessing applied before the tensorflow code.